src/components/container.js

```diff
@@ -188,4 +195,155 @@ export function createContainer(
       displayName || `Container(${Store.key.split('__')[0]})`;
     static hooks = { onInit, onUpdate, onCleanup };
   };
+
+  // eslint-disable-next-line no-unreachable
+  return createDynamicContainer({
+    matcher: (s) => s === Store,
+    onStoreInit: onInit,
+    onStoreCleanup: onCleanup,
+    onPropsUpdate: onUpdate,
+    displayName: displayName || `Container(${Store.key.split('__')[0]})`,
+  });
+}
+
+function useRegistry(scope, isGlobal, { globalRegistry = defaultRegistry }) {
+  return useMemo(() => {
+    const isLocal = !scope && !isGlobal;
+    return isLocal ? new StoreRegistry('__local__') : globalRegistry;
+  }, [scope, isGlobal, globalRegistry]);
+}
+
+function useContainedStore(scope, registry, hookActions, restProps) {
+  // Store contained scopes in a map, but throwing it away on scope change
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const containedStores = useMemo(() => new Map(), [scope]);
+
+  // Store props in a ref to avoid re-binding actions when they change
+  // If devs want to update consumers on prop change, they can put them in store
+  const containerProps = useRef();
+  containerProps.current = restProps;
+
+  const getContainedStore = useCallback(
+    (Store) => {
+      let containedStore = containedStores.get(Store);
+      // first time it gets called we add store to contained map bound
+      // so we can provide props to actions (only triggered by children)
+      if (!containedStore) {
+        const isExisting = registry.hasStore(Store, scope);
+        const { storeState } = registry.getStore(Store, scope);
+        const getProps = () => containerProps.current;
+        const actions = bindActions(Store.actions, storeState, getProps);
+        const hooks = bindActions(hookActions, storeState, getProps, actions);
+        containedStore = {
+          storeState,
+          actions,
+          hooks,
+          unsubscribe: storeState.subscribe(() => hooks.onStoreUpdate(Store)),
+        };
+        containedStores.set(Store, containedStore);
+        // signal store is contained and ready now, so by the time
+        // consumers subscribe we already have updated the store (if needed)
+        if (!isExisting) hooks.onStoreInit(Store);
+      }
+      return containedStore;
+    },
+    [scope, containedStores, registry, hookActions]
+  );
+  return [containedStores, getContainedStore];
+}
+
+// defaultRegistry.getStore
+
+function useApi(
+  memoMatcher,
+  getContainedStore,
+  { globalRegistry = defaultRegistry, getStore = defaultRegistry.getStore }
```

Comment:
Should it be?
```suggestion
  { globalRegistry = defaultRegistry, getStore = globalRegistry.getStore }
```

Reply:
Does not really matter much, as both are coming from context at the same time. These defaults are there because I'm worried enzyme shallow might not populate context properly and blow up. See https://github.com/atlassian/react-sweet-state/pull/57 
I might actually check if it is an issue with hooks

Reply:
Checked in product and `shallow` `enzyme` seems fine. Removed the default fallback.